### PR TITLE
Apply git clang-format to Detectors.

### PR DESCRIPTION
**Pull request by vkucera**

### CODE REVIEW

File: Detectors/EMCAL/base/src/EMCALBaseLinkDef.h [hunk 1]
```diff
@@ -23,5 +23,5 @@
 #pragma link C++ class o2::emcal::ClusterFactory < o2::emcal::Cell> + ;
 #pragma link C++ class o2::emcal::ClusterFactory < o2::emcal::Digit> + ;
 
-#pragma link C++ class std::vector < o2::emcal::Hit > + ;
+#pragma link C++ class std::vector < o2::emcal::Hit> + ;
```
Comment (sawenzel) [inline Detectors/EMCAL/base/src/EMCALBaseLinkDef.h:26]: really??

**Comment by sawenzel** [comment]
@vkucera : What is the motivation for this campaign? Was this discussed somewhere?

So far, our policy was to do correct formatting adiabatically and avoid pure formatting commits and my opinion is
that we could leave the CI resources for other important developments. So far, this approach has worked ok I believe.

**Comment by vkucera** [comment]
Hi @sawenzel , the motivation is to make sure that the existing code follows the current formatting style with the current clang-format version (which has been upgraded recently).

I have discussed this with @davidrohr who helped me to understand the details of the formatting checks. He may comment here, but I understood that it is acceptable to reformat the repository when there is a change in the formatting checks.

The issue with the current CI check is that it considers only changed lines which means that cases of improper formatting are piling up without ever being fixed. To fix that, one has to format the entire files from time to time.

As a compromise between a single huge PR, that would be difficult to review and merge, and many little PRs, that would trigger many testing builds, I split the changes into several PRs grouping the files in the same directories, that share the same code owners.
But if you have a suggestion how to do it better, I am open to a discussion and opinions from other reviewers as well.

**Comment by davidrohr** [comment]
For reference, I don't think one has to reformat the whole repository. This happens anyway adiabatically. Basically users should anyway run clang-format on their commit before opening a PR, so it happens actually automatically. But I also don't really mind few formatting commits from time to time. But to my knowledget there was no formal decision to reformat everything.

**Comment by shahor02** [comment]
Not sure how such PR can be reviewed. If we decide to keep the repo. compliant with evolving clang-format, then it is better to make a 1 global PR. In any case, this will certainly create some conflicts in user developments branches.
And I also don't understand the logic of changes like
````
- #pragma link C++ class std::vector < o2::emcal::Hit > + ;
+ #pragma link C++ class std::vector < o2::emcal::Hit> + ;
````

**Comment by vkucera** [comment]
> For reference, I don't think one has to reformat the whole repository. This happens anyway adiabatically. Basically users should anyway run clang-format on their commit before opening a PR, so it happens actually automatically. But I also don't really mind few formatting commits from time to time. But to my knowledget there was no formal decision to reformat everything.

I think that users usually apply `git clang-format` only to the changes in their commits, as instructed by the guidelines, instead of applying it to the entire files in their commits. With this approach the code will never be formatted properly.

If it is desirable to avoid the need for this kind of formatting PRs, users should be instructed clearly in the official guidelines to apply the formatting to the files instead of to the differences and the CI test should be modified accordingly.
Otherwise O2 will eventually turn into a mess of files consisting of pieces of code formatted in different ways.

**Comment by davidrohr** [comment]
But this I do not understand. Actually if all commits are git-clang-formatted beforehand, the repository remains consistent. It can only become inconsistent if
- PRs are merged despite the clang-format check failing.
- We change the clang-format rule or the clang version.

The current discrepancies I am seeing seem to be mostly comment indentation, and I believe this is due to a bumb in the clang version.

**Comment by vkucera** [comment]
OK, maybe we need a central decision on this matters. I will contact the responsible persons.

**Pull request closed without merging**